src/app/api/v1/user/email/route.ts

```diff
@@ -0,0 +1,142 @@
+  removeOneSecondaryEmail,
+  getEmailById,
+} from "../../../../../db/tables/emailAddresses.js";
+
+import { sendVerificationEmail } from "../../../utils/email";
+
+import { validateEmailAddress } from "../../../../../utils/emailAddress";
+import { getL10n } from "../../../../functions/server/l10n";
+const l10n = getL10n();
+
+interface EmailAddRequest {
+  email: string;
+}
+
+export async function POST(req: NextRequest) {
+  const token = await getToken({ req });
+
+  if (token) {
+    try {
+      const body: EmailAddRequest = await req.json();
+      const subscriber = await getSubscriberByEmail(token.email);
+      const emailCount = 1 + (subscriber.email_addresses?.length ?? 0); // primary + verified + unverified emails
+      const validatedEmail = validateEmailAddress(body.email);
+
+      if (validatedEmail === null) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-invalid-email"),
+        });
+      }
+
+      if (emailCount >= AppConstants.MAX_NUM_ADDRESSES) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-too-many-emails"),
+        });
+      }
+
+      // checkForDuplicateEmail
+      const emailLowerCase = validatedEmail.email.toLowerCase();
+      if (emailLowerCase === subscriber.primary_email.toLowerCase()) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-duplicate-email"),
+        });
+      }
+
+      for (const secondaryEmail of subscriber.email_addresses) {
+        if (emailLowerCase === secondaryEmail.email.toLowerCase()) {
+          return NextResponse.json({
+            success: false,
+            status: 400,
+            message: l10n.getString("user-add-duplicate-email"),
+          });
+        }
+      }
+
+      const unverifiedSubscriber = await addSubscriberUnverifiedEmailHash(
+        subscriber,
+        validatedEmail.email
+      );
+
+      await sendVerificationEmail(subscriber, unverifiedSubscriber.id);
+
+      return NextResponse.json({
+        success: true,
+        newEmailCount: emailCount + 1,
+        message: "Sent the verification email",
+      });
+      return NextResponse.json("ok");
+    } catch (e) {
+      console.error(e);
+      if (e.message === "error-email-validation-pending") {
+        return NextResponse.json(
+          {
+            success: false,
+            message: "Verification email recently sent, try again later",
+          },
+          { status: 429 }
+        );
+      }
+      return NextResponse.json({ success: false }, { status: 500 });
+    }
+  } else {
+    // Not Signed in, redirect to home
+    return NextResponse.redirect(AppConstants.SERVER_URL, 301);
+  }
+}
+
+interface EmailDeleteRequest {
+  emailId: string;
+}
+
+export async function DELETE(req: NextRequest) {
```

Comment:
While `DELETE` makes more sense, the actual API the frontend was using here was a `POST` call to `/remove-email` :unamused:

Reply:
Right, I was going to make that change on the frontend side (to use DEL) but up to you

Reply:
Oh yeah that works too. I was trying to modify as little as possible now, but let's at least reconsider this when turning this into proper React. Should be a fairly minimal change anyway.